src/ApiClient.ts

```diff
@@ -246,9 +253,14 @@ export default class ApiClient {
             'Stream connection closed. Resubscribing',
             err.toString()
           )
+
           if (new Date().getTime() - startTime < 1000) {
             await sleep(1000)
           }
+
+          if (onConnectionLost) {
+            onConnectionLost()
+          }
```

Comment:
i'm not super familiar with this code, but curious why the callback is called after sleeping?

Reply:
The reason is that if we are sleeping (to avoid slamming the network), we want the caller to be able to fill in the message history from a time closer to when the reconnect happens. Otherwise you could handle the `onConnectionLost` by filling in missing messages by querying and still have a second missing.

Reply:
Effectively, the time it spends sleeping is all part of the time the connection was lost.